### ISSUE TITLE
[ISSUE] Issue with `databricks_mws_log_delivery` resource provider version 1.25.0 ( databricks on AWS ) `cannot reset nil reader` error

### ISSUE THREAD

**Issue by michaeljmc** [databricks/terraform-provider-databricks#2675]
Issue when running a terraform plan against an existing log devliery location with 1.25.0 doesn't appear to fail when provider set to version = "1.24.1". work around set for now to use version = "1.24.1" but would like this resolved for future use. Thanks! Mike

### Configuration
```hcl
"databricks_mws_log_delivery" "your_unique_resource_name_here" {
  account_id               = youraccoundidhere
  credentials_id           = yourcredentials_id
  storage_configuration_id = yourstorage_configuration_id
  delivery_path_prefix     = "billable-usage"
  config_name          = "YourConfigNameHere"
  log_type             = "BILLABLE_USAGE"
  output_format        = "CSV"
  provider             = databricks.mws
  workspace_ids_filter = [your_workspace_id]
}

```

### Expected Behavior
no changes detected 

### Actual Behavior
Planning failed. Terraform encountered an error while generating this plan.

│ Error: cannot read mws log delivery: cannot reset nil reader

### Steps to Reproduce
execute a plan against a previously configured databricks_mws_log_delivery resource using provider version 1.24.1 or lower

### Terraform and provider versions
v1.2.9 and/or 1.4.6
1.24.1 does work but it doesn't work with 1.25.0

### Debug Output
not comfortable giving debug output at this point. but should be pretty easy to reproduce give instructions above and it appears to be a issue with the new provider version released today. 

### Important Factoids
if existing init(modules/provider etc) is in local directory it doesn't appear to be impacted. usual trouble shooting stuff but just wanted to be clear after a new init (-upgrade) is run it seems to no longer work properly.

**Comment by michaeljmc** [comment]
will update with details tomorrow . sorry didn't take my information i posted for some reason . details in subject should give some information though ...

**Comment by pietern** [comment]
Thanks for reporting the issue.

We're working on releasing a fix in 1.25.1 today.

**Comment by michaeljmc** [comment]
thanks pietern , the issue appears to be resolved with the 1.25.1 release. thank you!
